src/common/k8s-api/api-manager.injectable.ts

```diff
@@ -0,0 +1,256 @@
+/**
```

Comment:
This whole file could be made redundant by making the injectable API's also declare `setup()`. This would also comply to OCP nicely. Superficial, but very important.

Reply:
I can do that in a follow up PR